src/bridge/MemberListSyncer.ts

```diff
@@ -251,13 +250,13 @@ export class MemberListSyncer {
         return this.syncableRoomsPromise;
     }
 
-    private joinMatrixUsersToChannels(rooms: RoomInfo[], server: IrcServer, injectJoinFn: InjectJoinFn) {
+    private joinMatrixUsersToChannels(rooms: RoomInfo[], injectJoinFn: InjectJoinFn) {
         const d = promiseutil.defer();
 
         // filter out rooms listed in the rules
         const filteredRooms: RoomInfo[] = [];
         rooms.forEach((roomInfo) => {
-            if (!server.shouldSyncMembershipToIrc("initial", roomInfo.id)) {
+            if (!this.server.shouldSyncMembershipToIrc("initial", roomInfo.id)) {
```

Comment:
The constructor does not store `server` in `this.server`.

https://github.com/matrix-org/matrix-appservice-irc/blob/fb5cc2beac11c6758873fcbc8f534553b30a3cda/src/bridge/MemberListSyncer.ts#L53

Is this some language "feature" to silently add unused parameters of the constructor to the object?

Reply:
That's exactly what that does :). `private|public` before a parameter makes it a class property and stores it. Pretty handy